federation/pkg/federation-controller/ingress/ingress_controller.go

```diff
@@ -68,35 +91,41 @@ type IngressController struct {
 
 	// Backoff manager for ingresses
 	ingressBackoff *flowcontrol.Backoff
+	// Backoff manager for configmaps
+	configMapBackoff *flowcontrol.Backoff
 
 	// For events
 	eventRecorder record.EventRecorder
 
 	ingressReviewDelay    time.Duration
+	configMapReviewDelay  time.Duration
 	clusterAvailableDelay time.Duration
 	smallDelay            time.Duration
 	updateTimeout         time.Duration
 }
 
 // NewIngressController returns a new ingress controller
 func NewIngressController(client federation_release_1_4.Interface) *IngressController {
+	glog.V(4).Infof("->NewIngressController V(4)")
```

Comment:
I realize that this function is way too long and in need of refactoring - will do in a followup PR.